app/views/users/new.html.haml

```diff
@@ -1,14 +1,8 @@
-= title_tag "Portal: Sign up!"
-- @user.password = @user.password_confirmation = nil
-= form_for @user do |form|
-  %div{:id => dom_id_for(@user, :form), :class =>"action_menu"}
-    .action_menu
-      .action_menu_header_left
-        .padded_content
-          %h3 Signup
-      .action_menu_header_right
-        %ul.menu
-          %li= submit_tag 'Sign up'
-  = form.error_messages
-  = render :partial => 'form', :locals => { :f => form }
-  = render :partial => 'password', :locals => { :f => form }
+.landing-container
+
+:javascript
+  if (Portal.currentUser.isAnonymous) {
+    PortalPages.renderSignupModal({});
```

Comment:
See more detailed comment above. 

For using SSO providers, in the call to renderSingupModal():
Instead of the empty object {} you would instead pass
' { oauthProviders: Portal.oauthProviders }`